src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -30,12 +34,107 @@ public class NodeAutoCompleteSearchViewModel : SearchViewModel
+            set
+            {
+                noRecommendationsOrLowConfidenceTitle = value;
+                RaisePropertyChanged(nameof(NoRecommendationsOrLowConfidenceTitle));
+            }
+        }
+
+        /// <summary>
+        /// The No Recommendations or Low Confidence message
+        /// </summary>
+        public string NoRecommendationsOrLowConfidenceMessage
+        {
+            get { return noRecommendationsOrLowConfidenceMessage; }
+            set
+            {
+                noRecommendationsOrLowConfidenceMessage = value;
+                RaisePropertyChanged(nameof(NoRecommendationsOrLowConfidenceMessage));
+            }
+        }
+
+        /// <summary>
+        /// The Low Confidence additonal message
+        /// </summary>
+        public string LowConfidenceMessageAdditional
+        {
+            get { return lowConfidenceMessageAdditional; }
+            set
+            {
+                lowConfidenceMessageAdditional = value;
+                RaisePropertyChanged(nameof(LowConfidenceMessageAdditional));
+            }
+        }
+
+        /// <summary>
+        /// Indicates if display the No recommendations / Low confidence message (image and texts)
+        /// </summary>
+        public bool DisplayNoRecommendationsLowConfidence
+        {
+            get { return displayNoRecommendationsLowConfidence; }
+            set
+            {
+                displayNoRecommendationsLowConfidence = value;
+                RaisePropertyChanged(nameof(DisplayNoRecommendationsLowConfidence));
+            }
+        }
+
+        /// <summary>
+        /// Indicates if display the Low confidence option and Tooltip
+        /// </summary>
+        public bool DisplayLowConfidence
+        {
+            get { return displayLowConfidence; }
+            set
+            {
+                displayLowConfidence = value;
+                RaisePropertyChanged(nameof(DisplayLowConfidence));
+            }
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="dynamoViewModel">Dynamo ViewModel</param>
         internal NodeAutoCompleteSearchViewModel(DynamoViewModel dynamoViewModel) : base(dynamoViewModel)
         {
             // Off load some time consuming operation here
             InitializeDefaultAutoCompleteCandidates();
         }
 
+        /// <summary>
+        /// Reset Node AutoComplete search view state
+        /// </summary>
+        internal void ResetAutoCompleteSearchViewState()
+        {
+            if (!IsDisplayingMLRecommendation)
+            {
+                NoRecommendationsOrLowConfidenceMessage = string.Empty;
+                NoRecommendationsOrLowConfidenceTitle = string.Empty;
+                LowConfidenceMessageAdditional = string.Empty;
+                DisplayNoRecommendationsLowConfidence = false;
+                DisplayLowConfidence = false;
+            }
+            PopulateAutoCompleteCandidates();
+        }
```

Comment:
@jesusalvino Let me know if this is what you intend to reset the view, I would like to lazy call this function and not for each time when populating the results because that would be too frequent